### ISSUE TITLE
ShipAna.py produces empty histograms

### ISSUE THREAD

**Issue by JLTastet** [ShipSoft/FairShip#269]
On the latest master, following the instructions in the README (see below) results in empty histograms. I am not very familiar with what `ShipAna.py` is doing, but I remember seeing something non-empty in the past. So this looks like a regression. I have no idea where it could be though, and whether it matters at all. The MC truth and the event display look fine.

```bash
$ source /cvmfs/ship.cern.ch/SHiP-2020/latest/setUp.sh
$ alienv enter FairShip/latest
$> cd Workdir-HNL/
$> python2 "$FAIRSHIP"/macro/run_simScript.py
$> python2 "$FAIRSHIP"/macro/ShipReco.py -f ship.conical.Pythia8-TGeant4.root -g geofile_full.conical.Pythia8-TGeant4.root
$> python2 -i "$FAIRSHIP"/macro/ShipAna.py -f ship.conical.Pythia8-TGeant4_rec.root -g geofile_full.conical.Pythia8-TGeant4.root
$> python2 -i "$FAIRSHIP"/macro/eventDisplay.py -f ship.conical.Pythia8-TGeant4_rec.root -g geofile_full.conical.Pythia8-TGeant4.root
```

![fitresults](https://user-images.githubusercontent.com/8004066/63953605-f31f0880-ca81-11e9-800a-345718191d4d.gif)
![fitresults2](https://user-images.githubusercontent.com/8004066/63953620-f9ad8000-ca81-11e9-837f-621ea3bdea39.gif)
![fitresults2_pi0](https://user-images.githubusercontent.com/8004066/63953627-fca87080-ca81-11e9-98cb-7f3b11c64463.gif)

**Comment by ThomasRuf** [comment]
I know. It is an issue with the tank, it misses vacuum. Switch off check of fiducial volume, then histos will be filled. Thomad

Holen Sie sich Outlook für Android<https://aka.ms/ghei36>




On Thu, Aug 29, 2019 at 5:31 PM +0200, "Jean-Loup Tastet" <notifications@github.com<mailto:notifications@github.com>> wrote:


On the latest master, following the instructions in the README (see below) results in empty histograms. I am not very familiar with what ShipAna.py is doing, but I remember seeing something non-empty in the past. So this looks like a regression. I have no idea where it could be though, and whether it matters at all. The MC truth and the event display look fine.

$ source /cvmfs/ship.cern.ch/SHiP-2020/latest/setUp.sh
$ alienv enter FairShip/latest
$> cd Workdir-HNL/
$> python2 "$FAIRSHIP"/macro/run_simScript.py
$> python2 "$FAIRSHIP"/macro/ShipReco.py -f ship.conical.Pythia8-TGeant4.root -g geofile_full.conical.Pythia8-TGeant4.root
$> python2 -i "$FAIRSHIP"/macro/ShipAna.py -f ship.conical.Pythia8-TGeant4_rec.root -g geofile_full.conical.Pythia8-TGeant4.root
$> python2 -i "$FAIRSHIP"/macro/eventDisplay.py -f ship.conical.Pythia8-TGeant4_rec.root -g geofile_full.conical.Pythia8-TGeant4.root

[fitresults]<https://user-images.githubusercontent.com/8004066/63953605-f31f0880-ca81-11e9-800a-345718191d4d.gif>
[fitresults2]<https://user-images.githubusercontent.com/8004066/63953620-f9ad8000-ca81-11e9-837f-621ea3bdea39.gif>
[fitresults2_pi0]<https://user-images.githubusercontent.com/8004066/63953627-fca87080-ca81-11e9-98cb-7f3b11c64463.gif>

**Comment by JLTastet** [comment]
Thanks @ThomasRuf for the quick answer!
If it is a known issue which only affects this script, then it is fine. I just wanted to make sure that something else was not broken.

**Comment by JLTastet** [comment]
Solved / worked around by https://github.com/ShipSoft/FairShip/commit/244043f739d79f49b1a2465cf8d84369bc16f5d0.

**Comment by ThomasRuf** [comment]
It is not really solved, the original problem is still present. The vacuum tank is filled with air and not with vacuum. For this reason, the checking if decay is in fiducial volume fails. In order to fill the histograms nevertheless, the checking is not performed.

**Comment by JLTastet** [comment]
Sure. But we understand where the issue comes from, and it is not a bug in ShipAna.py itself, so it will be solved eventually.

**Comment by olantwin** [comment]
Maybe we could open an issue for the underlying problem?

The symptom is fixed by the work-around, but it's important that we keep track of the fact that it's only a work-around.

**Comment by JLTastet** [comment]
Makes sense. I can reopen this issue, so we don't forget to revert the workaround later once the underlying problem is fixed.

**Comment by olantwin** [comment]
Did we ever fix this?

Maksym had some issues with the fiducial volume cut (which I think we found a workaround for), but maybe this is related?
In any case, this is definitely one of the differences from the past analysis of muon DIS.

**Comment by olantwin** [comment]
@maksymovchynnikov Could you please make a PR for your changes to the UBT efficiency and fiducial volume cuts and veto decision?

If would be a good base for @anupama-reghunath to work from for the integration of the AdvSBT, and would also make sure that @yabezsh , you and @anupama-reghunath are all using the same definitions.

And hopefully we could then also finally close this issue for good.

**Comment by fairlyons** [comment]
On the current master the last set of histograms (relating to pi0) remains empty even with the fiducual volume check set to false.

**Comment by olantwin** [comment]
@fairlyons I think this is expected in events without a $pi^0$ (e.g. the default HNL decay channels). The other histograms are filled?

**Comment by fairlyons** [comment]
Yes all others have entries.
